File: src/main/java/ru/citeck/idea/codeinsight/forms/components/DateTime.java
package ru.citeck.idea.codeinsight.forms.components;

import lombok.Getter;

@Getter
public class DateTime extends PropertyComponent {

    public static final String TYPE = "datetime";
    private final DatePicker datePicker = new DatePicker();
    private final String format = "dd.MM.yyyy";
    private final Boolean enableTime = false;
    private final Boolean enableDate = true;

    @Override
    public String getType() {
        return TYPE;
    }

    @Getter
    static class DatePicker {
        String minDate = "";
        String maxDate = "";
    }

}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/DateTime.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

class DateTime : PropertyComponent() {

    companion object {
        const val TYPE = "datetime"
    }

    val datePicker = DatePicker()
    val format = "dd.MM.yyyy"
    val enableTime = false
    val enableDate = true

    class DatePicker {
        var minDate: String = ""
        var maxDate: String = ""
    }

    override fun getType(): String {
        return TYPE
    }
}
